### ISSUE TITLE
usedCoordinates Breaks Configuration Cache

### ISSUE THREAD

**Issue by johnfelenacorns** [apollographql/apollo-kotlin#4565]
### Summary

After modularizing some `.graphql` calls into its own module, other devs that pull the merged changes will have a configuration cache will fail because it can't find `usedCoordinates.json` of the new module

### Version

3.7.0

### Steps to reproduce the behavior

Reproduction steps:
1. Move some `graphql` files to its own module
2. Add `apolloSchema` to that module's `build.gradle`
3. Add that module to the schema module's `apolloUsedCoordinates` in `build.gradle`
4. Other devs do not have the `usedCoordinates.json` needed and fail at the configuration cache step

~~Sometimes~~ Steps To Fix:
1. Remove `org.gradle.unsafe.configuration-cache=true`
2. building
3. adding `org.gradle.unsafe.configuration-cache=true` back in
4. building again works

~~Other times:
We have had to also `rm -rf .gradle/configuration-cache` and `rm -rf ~/.gradle` before step 3~~

### Logs

```
<fullpath>/build/generated/usedCoordinates/apollo/service/usedCoordinates.json (No such file or directory)
```

### Anything else?

Not sure if https://github.com/apollographql/apollo-kotlin/pull/4564 is related

**Comment by martinbonnin** [comment]
Thanks for sending this! That stuff is complicated indeed as it configures multiples projects at once. I was expecting it to break project isolation but not necessarily configuration cache... I'll look into this!

**Comment by johnfelenacorns** [comment]
1. Correct, it doesn't need to be other devs.  At the end of last week, I can confirm it can be recreated by adding a new `apolloUsedCoordinates` entry and then trying to rebuild.  It was a valid entry where some `.graphql` files were moved from the `:schema` module to another module

2. I'm not sure if the second point happens, I will look out for it and get back to you if I can confirm either way

We are NOT using any remote configuration or build cache.  Our `gradle` files are in `Groovy`, which probably isn't an issue but just in case 


Futhremore, these steps seem to fix it every time now(I will strike-through the other steps in the Issue body):
1. Remove org.gradle.unsafe.configuration-cache=true
2. building
3. adding org.gradle.unsafe.configuration-cache=true back in
4. building again works

**Comment by johnfelenacorns** [comment]
So initially I thought it was just 1-time thing that every dev would have to go through.  I should have tested more after the original issue, my bad 😅 

The isuse happens until you turn off the configuration cache, sync, and build the project again.  After that, the configuration cache works again

I have been trying things this morning and can give a very specific set of reproduction steps now:
1. Add `project(':newmodule')` to `apolloUsedCoordinates` in your `:shcema`'s `build.gradle`.  It does NOT matter if this new module has `.graphql` files or ever has a file path to match `apollo.packageName`
```gradle
//schema build.gradle
apolloUsedCoordinates project(':somemodule'),
        project(':newmodule'),  // we just added this file
        project(':someothermodule')
```
2. Update your `:newmodule`'s `build.gradle` with needed `apollo` lines
```gradle
//newmodule build.gradle
apply plugin: "com.apollographql.apollo3"

apollo.packageName = "com.acorns.newmodule"

dependencies {
    apolloMetadata project(':schema')
    apolloSchema project(':schema')
}
```
3. Run the app, you should get the below error
```
/<file-path-to-project-root>/newmodule/build/generated/usedCoordinates/apollo/service/usedCoordinates.json (No such file or directory)
```

**Comment by johnfelenacorns** [comment]
Awesome, great news it is reproducible for you too @martinbonnin!

**Comment by martinbonnin** [comment]
Thanks for your patience with this. Fix is in https://github.com/apollographql/apollo-kotlin/pull/4578/files. I'm quite confident it'll fix your issue too because it looks really similar. 
We're planning a new version for Monday. If you have time to check the SNAPSHOTs when they're available (in ~2h) we'll know for sure. If not we'll cross fingers it was the same issue (or do another patch release)!

**Comment by johnfelenacorns** [comment]
Awesome thank you!

I will try it out on the release on Monday and report back

**Comment by BoD** [comment]
The fix is in v3.7.3 which has just been released.

**Comment by johnfelenacorns** [comment]
It looks like it works!  Thank you @martinbonnin 😀 

Sorry I got delayed with work and the holidays.  We updated our Apollo version to 3.7.3 and I couldn't reproduce the issue with the steps above
